# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/Loader.cpp
# Commit: baa464d5f255
# Full Hash: baa464d5f25599d4644ac371593d8cfec300d349
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2024-10-02 09:50:09
# Regressor Bug: 1899734
# File Overlap Count: 1
# Description:
#   Bug 1899734 - Part 4: Add SheetLoadData::InitiatorTypeString method. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D223350
# ==============================================================================

diff -r 432c1fa3fbcc -r baa464d5f255 layout/style/Loader.cpp
--- a/layout/style/Loader.cpp	Tue Oct 01 12:17:20 2024 +0000
+++ b/layout/style/Loader.cpp	Tue Oct 01 12:17:21 2024 +0000
@@ -1540,9 +1540,9 @@
     // Set the initiator type
     if (nsCOMPtr<nsITimedChannel> timedChannel =
             do_QueryInterface(httpChannel)) {
+      timedChannel->SetInitiatorType(aLoadData.InitiatorTypeString());
+
       if (aLoadData.mParentData) {
-        timedChannel->SetInitiatorType(u"css"_ns);
-
         // This is a child sheet load.
         //
         // The resource timing of the sub-resources that a document loads
@@ -1573,11 +1573,6 @@
           // report the resource.
           timedChannel->SetReportResourceTiming(false);
         }
-
-      } else if (aEarlyHintPreloaderId) {
-        timedChannel->SetInitiatorType(u"early-hints"_ns);
-      } else {
-        timedChannel->SetInitiatorType(u"link"_ns);
       }
     }
   }
@@ -1916,6 +1911,22 @@
   return LoadSheetResult{completed, isAlternate, matched};
 }
 
+nsLiteralString SheetLoadData::InitiatorTypeString() {
+  MOZ_ASSERT(mURI, "Inline sheet doesn't have the initiator type string");
+
+  if (mPreloadKind == StylePreloadKind::FromEarlyHintsHeader) {
+    return u"early-hints"_ns;
+  }
+
+  if (mParentData) {
+    // @import rule.
+    return u"css"_ns;
+  }
+
+  // <link>.
+  return u"link"_ns;
+}
+
 Result<Loader::LoadSheetResult, nsresult> Loader::LoadStyleLink(
     const SheetInfo& aInfo, nsICSSLoaderObserver* aObserver) {
   MOZ_ASSERT(aInfo.mURI, "Must have URL to load");